packages/csv-export/src/test/index.js

```diff
@@ -31,6 +31,29 @@ describe( 'generateCSVDataFromTable', () => {
 			mockCSVData
 		);
 	} );
+
+	it( 'should prefix tab character when the cell value starts with one of =, +, -, and @', () => {
+		[ '=', '+', '-', '@' ].forEach( ( val ) => {
+			const expected = 'value\n\t' + val + 'test';
+			const result = generateCSVDataFromTable(
+				[
+					{
+						label: 'value',
+						key: 'value',
+					},
+				],
+				[
+					[
+						{
+							display: 'value',
+							value: val + 'test',
+						},
+					],
+				]
+			);
+			expect( result ).toBe( expected );
+		} );
+	} );
```

Comment:
Thank you for this. ❤️

Reply:
Added changelog and updated code to surround cells with a double 
quotation.

I'm also including CSV files for testing purposes.

One thing to note is a cell with a negative number is aligned to the left since it is treated as a string. You can check it with `with-fix-number.csv` file.

[with-fix.csv](https://github.com/woocommerce/woocommerce-admin/files/6634803/with-fix.csv)
[with-fix-number.csv](https://github.com/woocommerce/woocommerce-admin/files/6634804/with-fix-number.csv)
[without-fix.csv](https://github.com/woocommerce/woocommerce-admin/files/6634805/without-fix.csv)

Reply:
>One thing to note is a cell with a negative number is aligned to the left since it is treated as a string

Can you please share a screenshot, with numbers on mac they are aligned to the right. 🤔 

![image](https://user-images.githubusercontent.com/15019298/122291654-ae908580-cf12-11eb-93c9-ecea888f91b2.png)


Also I've tested it once again, it is working well with google sheets and numbers.

![image](https://user-images.githubusercontent.com/15019298/122291540-8f91f380-cf12-11eb-9327-1c0ac313e9de.png)

Reply:
@naman03malhotra Thank you for the review 👍 

My apologies for the confusion. The left-aligned number only appears on Microsoft Excel, not on Google Sheets.

Microsft Excel also treats the cell as a string with the tab char included as you see on the screenshot. 

Sceenshot:

![Screen Shot 2021-06-16 at 7 07 39 PM](https://user-images.githubusercontent.com/4723145/122319951-40c66880-ced6-11eb-9cb3-e5068e921eb0.jpg)